src/MCPClient/lib/clientScripts/rightsFromCSV.py

```diff
@@ -0,0 +1,381 @@
+            self.object_basis_act_usage[filepath][basis][act] = True
+        else:
+            print "Skipping duplicate basis/act combination at row", self.rows_processed + 1
+
+        self.rows_processed += 1
+
+    def exit_with_error(self, message, value=None):
+        """Print message to standard error and exit with status 1."""
+        message = '[Row {}] {}'.format(self.rows_processed + 1, message)
+        if value:
+            print >>sys.stderr, message, value
+        else:
+            print >>sys.stderr, message
+        exit(1)
+
+    def store_row(self):
+        """Create rights records in database using row data."""
+        rights_statement = self.generate_rights_statement()
+
+        if rights_statement.rightsbasis == 'Copyright':
+            self.store_copyright_info(rights_statement)
+
+        elif rights_statement.rightsbasis == 'License':
+            self.store_license_info(rights_statement)
+
+        elif rights_statement.rightsbasis == 'Statute':
+            self.store_statute_info(rights_statement)
+
+        elif rights_statement.rightsbasis in ['Other', 'Donor', 'Policy']:
+            self.store_other_info(rights_statement)
+
+        self.store_grant_info(rights_statement)
+
+    def generate_rights_statement(self):
+        """Generate rights statement."""
+        basis = self.column_value('basis').lower().capitalize()
+
+        if basis not in dict(models.RightsStatement.RIGHTS_BASIS_CHOICES):
+            self.exit_with_error('Invalid basis:', basis)
+
+        # Get file data
+        filepath = self.column_value('file')
+        transfer_file = models.File.objects.filter(originallocation='%transferDirectory%' + filepath, transfer_id=self.transfer_uuid).first()
+
+        if (transfer_file is None):
+            self.exit_with_error('Record not found for file', filepath)
+
+        # Create rights statement
+        rights_statement = models.RightsStatement()
+        rights_statement.metadataappliestotype = self.metadata_applies_to_type
+        rights_statement.metadataappliestoidentifier = transfer_file.uuid
+        rights_statement.rightsbasis = basis
+        rights_statement.status = 'ORIGINAL'
+        rights_statement.save()
+
+        return rights_statement
+
+    def validate_columns(self, row):
+        """Check for invalid/missing columns."""
+        for column_name in row.keys():
+            if not column_name in self.allowed_column_names:
+                self.exit_with_error('Invalid column found:', column_name)
+
+        for column_name in self.required_column_names:
+            if not column_name in row:
+                self.exit_with_error('Missing required column:', column_name)
+
+    def column_value(self, column_name):
+        """Return value of a row column by it's name in the header, None if missing/blank."""
+        if column_name in self.current_row:
+            value = self.current_row[column_name]
+            return None if value.strip() == '' else value
```

Comment:
Sorry I was unclear before - you don't need the `if column_name in self.current_row:`.  `.get` is like `dict[index]`, except if the entry isn't there it doesn't raise an exception, but returns the second parameter instead.